components/hooks/useDecimalInput.ts

```diff
@@ -0,0 +1,21 @@
+// hooks/useDecimalInput.ts
+
+import { useState, ChangeEvent } from 'react';
+import Decimal from 'decimal.js';
+
+const handleDecimalInputChange = (
+  event: ChangeEvent<HTMLInputElement>,
+  setValue: React.Dispatch<React.SetStateAction<Decimal>>
+) => {
+  const { value } = event.target;
+  setValue(new Decimal(value));
+};
+
+const useDecimalInput = (initialValue: Decimal = new Decimal(0)) => {
```

Comment:
i see that you love duplication , why you didnt use useFormInput 

const useDecimalInput = (initialValue: Decimal = new Decimal(0)) => {
  const [value, handleChange] = useInput({ value: initialValue }, value => new Decimal(value));
  return [value.value, handleChange] as const;
};